Assignment5/app.py

```
@@ -0,0 +1,184 @@
+
+
+def load_data() -> Dict[str, Any]:
+    """Load company dataset from disk."""
+    if not DATA_FILE.exists():
+        return {"companies": []}
+
+    try:
+        with DATA_FILE.open(encoding="utf-8") as fp:
+            return json.load(fp)
+    except json.JSONDecodeError as exc:
+        raise ValueError(f"Invalid JSON structure: {exc}") from exc
```

Ruff (0.14.4)

Avoid specifying long messages outside the exception class

(TRY003)